src/main/java/appeng/core/api/ApiPart.java

```diff
@@ -80,64 +90,79 @@ public void initFMPSupport()
 		}
 	}
 
+	/**
+	 * Conceptually this method will build a new class hierarchy that is rooted at the given base class, and includes a chain of all registered layers.
+	 * <p/>
+	 * To accomplish this, it takes the first registered layer, replaces it's inheritance from LayerBase with an inheritance from the given baseClass,
+	 * and uses the resulting class as the parent class for the next registered layer, for which it repeats this process. This process is then repeated
+	 * until a class hierarchy of all layers is formed. While janking out the inheritance from LayerBase, it'll make also sure that calls to that
+	 * classes method will instead be forwarded to the superclass that was inserted as part of the described process.
+	 * <p/>
+	 * Example: If layers A and B are registered, and TileCableBus is passed in as the baseClass, a synthetic class A_B_TileCableBus should be returned,
+	 * which has A_B_TileCableBus -extends-> B_TileCableBus -extends-> TileCableBus as it's class hierarchy, where A_B_TileCableBus has been generated
+	 * from A, and B_TileCableBus has been generated from B.
+	 */
 	public Class<? extends AEBaseTile> getCombinedInstance( final Class<? extends AEBaseTile> baseClass )
 	{
-		String base = baseClass.getName();
-
 		if( this.desc.isEmpty() )
 		{
+			// No layers registered...
 			return baseClass;
 		}
 
-		final String description = base + ':' + Joiner.on( ";" ).skipNulls().join( this.desc.iterator() );
+		return cache.getUnchecked( new CacheKey( baseClass, this.desc ) );
+	}
+
+	private Class<? extends AEBaseTile> generateCombinedClass( CacheKey cacheKey )
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Move this method into the anonymous class declared at line 71. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonar.orod.org/coding_rules#rule_key=squid%3AS3398)